pkg/csi/service/vanilla/controller.go

```diff
@@ -2642,9 +2643,18 @@ func (c *controller) processQueryResultsListVolumes(ctx context.Context, startin
 			nodeVMUUID, found := volumeIDToNodeUUIDMap[blockVolID]
 			if found {
 				volCounter += 1
-				//Populate csi.Volume info for the given volume
+				volumeId := blockVolID
+				migratedVolumePath, err := volumeMigrationService.GetVolumePath(ctx, blockVolID, false)
+				if err != nil && err == common.ErrNotFound {
```

Comment:
Since we need to make a decision here based on migration service in-memory cache, can you write a new method in migration service to return volumePath from cache for ex: GetVolumePathFromMigrationServiceCache(). That way, the error handling will be simpler.